File: frontend/src/features/masters/models/Question.ts
import type { MasterListParams, MasterRecord } from './common';

export type AnswerType =
  | 'Text Box'
  | 'None'
  | 'Dropdown (single select)'
  | 'Dropdown (multi select)'
  | 'Date & Time'
  | 'Masters';

export const ANSWER_TYPE_OPTIONS: { label: string; value: AnswerType }[] = [
  { label: 'None', value: 'None' },
  { label: 'Text Box', value: 'Text Box' },
  { label: 'Dropdown (single select)', value: 'Dropdown (single select)' },
  { label: 'Dropdown (multi select)', value: 'Dropdown (multi select)' },
  { label: 'Date & Time', value: 'Date & Time' },
  { label: 'Masters', value: 'Masters' },
];

export interface Question extends MasterRecord {
  title: string;
  answer_type: AnswerType;
  has_text_box: boolean;
  has_multiple_text_box: boolean;
  has_sub_question: boolean;
  is_validation_required: boolean;
  has_response_option: boolean;
  allow_multiple_input: boolean;
  has_associated_master: boolean;
  is_calculated: boolean;
  validation_type_id: number | null;
  master_type: string | null;
  parent_question_id: number | null;
  sub_question_ids: number[];
}

export interface CreateQuestionRequest {
  title: string;
  answer_type: AnswerType;
  has_text_box: boolean;
  has_multiple_text_box: boolean;
  has_sub_question: boolean;
  is_validation_required: boolean;
  has_response_option: boolean;
  allow_multiple_input: boolean;
  has_associated_master: boolean;
  is_calculated: boolean;
  is_active: boolean;
  validation_type_id: number | null;
  master_type: string | null;
  parent_question_id: number | null;
  sub_question_ids: number[];
}

export type UpdateQuestionRequest = Partial<CreateQuestionRequest>;
export type QuestionListParams = MasterListParams;

File: backend/src/domain/entities/masters/question.py
"""Question Master — domain entity."""
from dataclasses import dataclass, field

from src.domain.entities.base_entity import BaseEntity
from src.domain.enums.answer_type_enum import AnswerType
from src.domain.enums.master_type_enum import MasterType


@dataclass
class Question(BaseEntity):
    title: str = field(default="")
    answer_type: AnswerType = field(default=AnswerType.NONE)
    has_text_box: bool = field(default=False)
    has_multiple_text_box: bool = field(default=False)
    has_sub_question: bool = field(default=False)
    is_validation_required: bool = field(default=False)
    has_response_option: bool = field(default=False)
    allow_multiple_input: bool = field(default=False)
    has_associated_master: bool = field(default=False)
    is_calculated: bool = field(default=False)
    is_active: bool = field(default=True)
    validation_type_id: int | None = field(default=None)
    parent_question_id: int | None = field(default=None)
    master_type: MasterType | None = field(default=None)
    sub_question_ids: list[int] = field(default_factory=list)
